### PR TITLE
extend matrix test and use env variables.

### PR DESCRIPTION
**Pull request by fredrikhgrelland**
This extends on #161 and uses the strategy of #160 in order to matrix test using environment variables.
Due to how github actions print the name of the step with matrix strategy I have made some changes to the logic.

Essentially github will create a matrix and print the value e.g. `test (allow)` in #161. This is not readable for multiple inputs: `test (true, allow)` This PR uses descriptive names for the matrix to make it readable.

### Process:
1. create matrix
2. create output variable
3. create env variable and put in file .env_override

Extention of the Makefile used for test will source .env_override after .env allowing us to have a default overridden by github action testing.

### CODE REVIEW

**Comment by fredrikhgrelland** [comment]
close #162

**Comment by pdmthorsrud** [comment]
This approach will be dependent on #160, that makes the variables used available inside the box. That PR has some problems, see https://github.com/fredrikhgrelland/vagrant-hashistack/pull/160#issuecomment-669869615

File: .github/action/create-env.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+out_file = ""
+
+for key in env_keys:
+    if key.startswith("ENVKEY_"):
+        out_file += key.split("ENVKEY_")[1] + "=" + os.environ.get(key) + "\n"
+
+with open( str(os.environ.get("GITHUB_WORKSPACE")) + "/" + str(os.environ.get("FILE_NAME")), "w") as text_file:
+    text_file.write(out_file)
+text_file.close()
```
Comment (Neha-Sinha2305) [inline .github/action/create-env.py:13]: Do we need this statement? We are using **with** keyword so may be this is not required.
Text from https://docs.python.org/3/tutorial/inputoutput.html
"It is good practice to use the with keyword when dealing with file objects. The advantage is that the file is properly closed after its suite finishes, even if an exception is raised at some point"

**Comment by pdmthorsrud** [comment]
> This approach will be dependent on #160, that makes the variables used available inside the box. That PR has some problems, see [#160 (comment)](https://github.com/fredrikhgrelland/vagrant-hashistack/pull/160#issuecomment-669869615)

#160 is now using a yaml-file to toggle ACLs in nomad, instead of a `.env`-file,.

File: .github/action/create-env.py (hunk 1 above)
Comment (fredrikhgrelland) [reply .github/action/create-env.py:13]: You are right. Cool if you fix that in the PR :+1:

**Pull request merged**